lib/ansible/vars/hostvars.py

```diff
@@ -111,6 +111,12 @@ def __repr__(self):
             out[host] = self.get(host)
         return repr(out)
 
+    def __deepcopy__(self, memo):
+        # We do not need to deepcopy because HostVars is immutable,
+        # however we have to implement the method so we can deepcopy
+        # variables' dicts that contain HostVars.
+        return self
+
```

Comment:
probably need same in hostvarsvars

Reply:
I don't think it is needed because `deepcopy()` on `HostVars` returns self and so `deepcopy()` does not continue further down the structure to be applied on `HostVarsVars`.